From 6ee776656c6476fb23dae472bdd5254bec3d9888 Mon Sep 17 00:00:00 2001
From: Ricardo Cerqueira <cyanogenmod@cerqueira.org>
Date: Wed, 18 Dec 2013 02:43:17 +0000
Subject: [PATCH] BatteryService: Restore the original parcel order

Append the dock extensions to the end of the parcel to preserve
compatibility with the original parcel format.

Change-Id: Ib4fa8473fcab9882d41bd9f9617af163154290ab
---
 include/batteryservice/BatteryService.h       | 2 +-
 services/batteryservice/BatteryProperties.cpp | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/include/batteryservice/BatteryService.h b/include/batteryservice/BatteryService.h
index e9376f4..105229d 100644
--- a/include/batteryservice/BatteryService.h
+++ b/include/batteryservice/BatteryService.h
@@ -45,7 +45,6 @@ enum {
 
 struct BatteryProperties {
     bool chargerAcOnline;
-    bool chargerDockAcOnline;
     bool chargerUsbOnline;
     bool chargerWirelessOnline;
 
@@ -59,6 +58,7 @@ struct BatteryProperties {
     int batteryTemperature;
     String8 batteryTechnology;
 
+    bool chargerDockAcOnline;
     bool dockBatterySupported;
     int dockBatteryStatus;
     int dockBatteryHealth;
diff --git a/services/batteryservice/BatteryProperties.cpp b/services/batteryservice/BatteryProperties.cpp
index 4111d9a..8c7d287 100644
--- a/services/batteryservice/BatteryProperties.cpp
+++ b/services/batteryservice/BatteryProperties.cpp
@@ -31,7 +31,6 @@ namespace android {
 
 status_t BatteryProperties::readFromParcel(Parcel* p) {
     chargerAcOnline = p->readInt32() == 1 ? true : false;
-    chargerDockAcOnline = p->readInt32() == 1 ? true : false;
     chargerUsbOnline = p->readInt32() == 1 ? true : false;
     chargerWirelessOnline = p->readInt32() == 1 ? true : false;
 
@@ -45,6 +44,7 @@ status_t BatteryProperties::readFromParcel(Parcel* p) {
     batteryTemperature = p->readInt32();
     batteryTechnology = String8((p->readString16()).string());
 
+    chargerDockAcOnline = p->readInt32() == 1 ? true : false;
     dockBatterySupported = p->readInt32() == 1 ? true : false;
     dockBatteryStatus = p->readInt32();
     dockBatteryHealth = p->readInt32();
@@ -61,7 +61,6 @@ status_t BatteryProperties::readFromParcel(Parcel* p) {
 
 status_t BatteryProperties::writeToParcel(Parcel* p) const {
     p->writeInt32(chargerAcOnline ? 1 : 0);
-    p->writeInt32(chargerDockAcOnline ? 1 : 0);
     p->writeInt32(chargerUsbOnline ? 1 : 0);
     p->writeInt32(chargerWirelessOnline ? 1 : 0);
 
@@ -75,6 +74,7 @@ status_t BatteryProperties::writeToParcel(Parcel* p) const {
     p->writeInt32(batteryTemperature);
     p->writeString16(String16(batteryTechnology));
 
+    p->writeInt32(chargerDockAcOnline ? 1 : 0);
     p->writeInt32(dockBatterySupported ? 1 : 0);
     p->writeInt32(dockBatteryStatus);
     p->writeInt32(dockBatteryHealth);
-- 
1.8.3.2

